### PR TITLE
Bugfix/741 kbs sometimes used even though they are disabled

### PR DESCRIPTION
**Pull request by MdeBoer95**
**What's in the PR**
* use only enabled KBs for annotations and the KB page
* warn the user if a disabled KB is set as scope for the concept feature

**How to test manually**
* check if warning is displayed properly when a disabled kb is set as scope for the concept feature
* check if bug has been fixed according to #741 

**Automatic testing**
* [x] PR includes unit tests

**Documentation**
* [ ] PR updates documentation

### CODE REVIEW

File: inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/ConceptFeatureTraitsEditor.java [hunk 1]
```diff
@@ -129,13 +133,18 @@ private Traits readTraits()
...
                 result.setKnowledgeBase(kb.get());
                 if (t.getScope() != null) {
                     kbService.readConcept(kb.get(), t.getScope(), true)
                             .ifPresent(concept -> result.setScope(KBHandle.of(concept)));
                 }
             }
+            else {
+                warn("The selected knowledge base for this Concept Feature has been removed or disabled. "
+                    + "Select a new knowledge base and click on save");
+                LOG.error("The selected knowledge base has been removed or disabled.");
```
[the remark this answers is not in the document]
Comment (MdeBoer95) [reply inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/ConceptFeatureTraitsEditor.java]: Does this always apply? I think we have multiple places in the code where we do something like:
error(...)  
LOG.error(...)

[the remark this answers is not in the document]
Comment (reckart) [reply inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/ConceptFeatureTraitsEditor.java]: Yes, that is correct. The reason is we include the exception in `LOG.error("some message", e)` which causes the stack trace to be logged. But we cannot include the exception (stack trace) with `error(...)`. So we log twice.

**Comment by MdeBoer95** [comment]
> The `createDisabledKbWarningLabel()` is repeated multiple times. Instead of copy/pasting the code, it would be good to extract it into an independent Wicket component and then to add that component where required.

Should this wicket component take an annotation feature in the constructor and set visibility, resolve name of the kb by itself or should the editor do this and just pass a warning message to the component?

**Comment by reckart** [comment]
> Should this wicket component take an annotation feature in the constructor and set visibility, resolve name of the kb by itself or should the editor do this and just pass a warning message to the component?

That is a good question. What are the pro's/con's from your perspective?

**Comment by MdeBoer95** [comment]
> That is a good question. What are the pro's/con's from your perspective?

The first would minimize code redundancy, but makes the component more specific and less flexible. Thus it can probably only be used in that exact context.

The second gives a  more general component and is a bit "cleaner" IMO but we will still have redundant code across the different editors.(For resolving the name and setting the visibility)

**Comment by reckart** [comment]
It sounds like the first would be a good choice.

**Comment by MdeBoer95** [comment]
@reckart 
https://github.com/inception-project/inception/blob/dfc0a3c9f22a815640f8f0e42eebd24b22243c4a/inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/ConceptFeatureTraits.java#L32-L87

The kbService is not being initialized. Hence I get a NullPointerException when I try to call ```isKbEnabled(...)```
Do you know what I need to do in order to initialize the service properly?

**Comment by reckart** [comment]
It is not possible to inject Spring beans into traits. Traits are data objects, not services (`@Component`).

**Comment by reckart** [comment]
The traits objects should be plain beans and not contain logic.

**Comment by reckart** [comment]
@MdeBoer95 is the PR still WIP?

**Comment by MdeBoer95** [comment]
yes

**Comment by MdeBoer95** [comment]
Now done.

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Pull request merged**